pyirf/tests/test_interpolation.py

```diff
@@ -31,6 +31,22 @@ def test_interpolate_effective_area_per_energy_and_fov():
     assert np.allclose(aeff_interp[:, 0], aeff0, rtol=0.03, atol=min_aeff)
 
 
+def calc_mean_std(matrix, vals):
+    """Auxiliary function to compute mean and std from 'matrix' along an axis which values are in 'values'."""
+    n_en = matrix.shape[0]
+    means = np.empty(n_en)
+    stds = np.empty(n_en)
+    for i_en in np.arange(n_en):
+        w = matrix[i_en, :]
+        if np.sum(w) > 0:
```

Comment:
For weights to make sense, they have to be all > 0, not just the sum

Reply:
actually there should be no negative weights, but it can happen at low energies that all bins are empty and you have 0s everywhere, this check is meant exactly for such a case